pkg/tekton/metapipeline/clientfactory.go

```diff
@@ -0,0 +1,295 @@
+// The responsibility of the meta pipeline is to prepare the execution pipeline and to allow Apps to contribute
+// the this execution pipeline.
+func NewMetaPipelineClient() (*ClientFactory, error) {
+	tektonClient, jxClient, kubeClient, ns, err := getClientsAndNamespace()
+	if err != nil {
+		return nil, err
+	}
+
+	return NewMetaPipelineClientWithClientsAndNamespace(jxClient, tektonClient, kubeClient, ns)
+}
+
+// NewMetaPipelineClientWithClientsAndNamespace creates a new client for the creation and application of meta pipelines using the specified parameters.
+func NewMetaPipelineClientWithClientsAndNamespace(jxClient versioned.Interface, tektonClient tektonclient.Interface, kubeClient kubernetes.Interface, ns string) (*ClientFactory, error) {
+	client := ClientFactory{
+		jxClient:       jxClient,
+		tektonClient:   tektonClient,
+		kubeClient:     kubeClient,
+		ns:             ns,
+		defaultImage:   defaultContainerImage,
+		serviceAccount: defaultServiceAccount,
+	}
+
+	return &client, nil
+}
+
+// Create creates the Tekton CRDs needed for executing the pipeline as defined by the input parameters
+func (c *ClientFactory) Create(pullRef PullRef, pipelineKind PipelineKind, context string, envs map[string]string, labels map[string]string) (kube.PromoteStepActivityKey, tekton.CRDWrapper, error) {
+	err := c.cloneVersionStreamIfNeeded()
+	if err != nil {
+		return kube.PromoteStepActivityKey{}, tekton.CRDWrapper{}, errors.Wrap(err, "unable to clone version stream")
+	}
+
+	gitInfo, err := gits.ParseGitURL(pullRef.SourceURL())
+	if err != nil {
+		return kube.PromoteStepActivityKey{}, tekton.CRDWrapper{}, errors.Wrap(err, fmt.Sprintf("unable to determine needed git info from the specified git url '%s'", pullRef.SourceURL()))
+	}
+
+	podTemplates, err := c.getPodTemplates(apps.AppPodTemplateName)
+	if err != nil {
+		return kube.PromoteStepActivityKey{}, tekton.CRDWrapper{}, errors.Wrap(err, "unable to retrieve pod templates")
+	}
+
+	branchIdentifier, err := c.determineBranchIdentifier(pipelineKind, pullRef)
+	if err != nil {
+		return kube.PromoteStepActivityKey{}, tekton.CRDWrapper{}, errors.Wrap(err, "unable to create branch identifier")
+	}
+
+	// resourceName is shared across all builds of a branch, while the pipelineName is unique for each build.
+	resourceName := tekton.PipelineResourceNameFromGitInfo(gitInfo, branchIdentifier, context, tekton.MetaPipeline, nil, "")
+	pipelineName := tekton.PipelineResourceNameFromGitInfo(gitInfo, branchIdentifier, context, tekton.MetaPipeline, c.tektonClient, c.ns)
+	buildNumber, err := tekton.GenerateNextBuildNumber(c.tektonClient, c.jxClient, c.ns, gitInfo, branchIdentifier, retryDuration, context)
+	if err != nil {
+		return kube.PromoteStepActivityKey{}, tekton.CRDWrapper{}, errors.Wrap(err, "unable to determine next build number")
+	}
+
+	logger.WithField("repo", gitInfo.URL).WithField("buildNumber", buildNumber).Info("creating meta pipeline CRDs ")
```

Comment:
shouldn't be this `debug`?

Reply:
I'd go for info. What are your concerns. Logs getting too big? So you would suggest that all the controller logs is that it started and then nothing more, except there is an error or warning? Just trying to understand where you are coming from?

Reply:
ahh, I guess now I understand why you want to use Debug logging. This is also called from `jx start pipeline` and from the CLI  I don't really want this to be visible per default.  

I only had the pipeline runner in mind as client when I wrote this where I would expect these type of things being logged per default. 

I'll change all logging in this file to Debug and then we just need to make sure we start the pipelinerunner per default in verbose mode.